File: test/languages/sample.dart
import 'dart:async';

enum ReviewState { draft, published }

sealed class Result<T> {
  const Result();
}

final class Success<T> extends Result<T> {
  const Success(this.value);
  final T value;
}

typedef Review = ({int id, String title, double? score});

Future<String> summarize(Review review) async {
  await Future<void>.delayed(Duration.zero);
  final (:id, :title, :score) = review;
  return switch (score) {
    null => '$id: $title is unrated',
    >= 8 => '$id: $title is recommended',
    _ => '$id: $title scored ${score.toStringAsFixed(1)}',
  };
}

void main() async => print(await summarize((id: 1, title: 'Night Walk', score: 8.5)));

File: test/languages/sample.jl
module AkihabaraFixture

export Review, summarize

@enum ReviewState draft published

struct Review{T<:Real}
    id::Int
    title::String
    score::Union{T,Nothing}
    state::ReviewState
end

function summarize(reviews::AbstractVector{<:Review}; limit::Integer=10)
    published_reviews = filter(review -> review.state == published, reviews)
    sorted = sort(published_reviews; by=review -> something(review.score, 0), rev=true)
    return ["$(review.title): $(round(something(review.score, 0); digits=1))"
            for review in Iterators.take(sorted, limit)]
end

macro fixture(expression)
    return :(println($(esc(expression))))
end

@fixture summarize([Review(1, "Night Walk", 8.5, published)])

end
